homeassistant/components/climate/myicomfort.py

```diff
@@ -0,0 +1,225 @@
+    STATE_ON, STATE_OFF, STATE_IDLE, ATTR_TEMPERATURE)
+
+REQUIREMENTS = ['myicomfort==0.2.0']
+
+_LOGGER = logging.getLogger(__name__)
+
+# HA doesn't have a 'circulate' state defined.
+STATE_CIRCULATE = 'circulate'
+
+SUPPORT_FLAGS = (SUPPORT_TARGET_TEMPERATURE |
+                 SUPPORT_TARGET_TEMPERATURE_HIGH |
+                 SUPPORT_TARGET_TEMPERATURE_LOW |
+                 SUPPORT_OPERATION_MODE |
+                 SUPPORT_AWAY_MODE |
+                 SUPPORT_FAN_MODE)
+
+FAN_MODES = [
+    STATE_AUTO, STATE_ON, STATE_CIRCULATE
+]
+
+OP_MODES = [
+    STATE_OFF, STATE_HEAT, STATE_COOL, STATE_AUTO
+]
+
+SYSTEM_STATES = [
+    STATE_IDLE, STATE_HEAT, STATE_COOL
+]
+
+TEMP_UNITS = [
+    TEMP_FAHRENHEIT, TEMP_CELSIUS
+]
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Optional('name', default='iComfort'): cv.string,
+    vol.Optional('system', default=0): vol.Coerce(int),
+    vol.Optional('zone', default=0): vol.Coerce(int),
+    vol.Optional('min_temp'): vol.Coerce(float),
+    vol.Optional('max_temp'): vol.Coerce(float),
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the climate platform."""
+    from myicomfort.api import Tstat
+
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+    system = config.get('system')
+    zone = config.get('zone')
+    name = config.get('name')
+    min_temp = config.get('min_temp')
+    max_temp = config.get('max_temp')
+
+    climate = [MyiComfortClimate(name, min_temp, max_temp,
+                                 Tstat(username, password, system, zone))]
+
+    add_devices(climate, True)
+
+
+class MyiComfortClimate(ClimateDevice):
+    """Class for Lennox iComfort WiFi thermostat."""
+
+    def __init__(self, name, min_temp, max_temp, api):
+        """Initialize the climate device."""
+        self._name = name
+        self._api = api
+        self._min_temp = min_temp
+        self._max_temp = max_temp
+
+    def update(self):
+        """Update data from the thermostat API."""
+        self._api.pull_status()
+
+    @property
+    def device_state_attributes(self):
+        """Return device specific state attributes."""
+        return {
+        }
+
+    @property
+    def state(self):
+        """Return the current operational state."""
+        return SYSTEM_STATES[self._api.state]
+
+    @property
+    def name(self):
+        """Return the name of the climate device."""
+        return self._name
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return SUPPORT_FLAGS
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        return TEMP_UNITS[self._api.temperature_units]
+
+    @property
+    def min_temp(self):
+        """Return the minimum temperature."""
+        if self._min_temp:
```

Comment:
Check for `None` instead. The temperature could be 0 which would eval as `False`.